src/DataGrid.tsx

```diff
@@ -857,11 +858,11 @@ function DataGrid<R, K extends keyof R, SR>({
       startRowIndex++;
       let key: string | number = hasGroups ? startRowIndex : rowIdx;
       let isRowSelected = false;
-      if (rowKey !== undefined) {
-        const rowId = row[rowKey];
-        isRowSelected = selectedRows?.has(rowId) ?? false;
-        if (typeof rowId === 'string' || typeof rowId === 'number') {
-          key = rowId;
+      if (typeof rowKeyGetter === 'function') {
+        const rowKey = rowKeyGetter(row);
+        isRowSelected = selectedRows?.has(rowKey) ?? false;
+        if (typeof rowKey === 'string' || typeof rowKey === 'number') {
```

Comment:
Should we still manually typecheck this?

Reply:
Probably not as we are using `React.Key`